url529c09a30c_kianyangchn_DriveNow_tgz-pJ8-drivenow_tests_TestJ8/src/drivenow/controls/Ressourcenverwaltung.java

```diff
@@ -0,0 +1,326 @@
+	 * Eine entsprechende Fehlermeldung ist auszugeben. Existiert mindestens
+	 * eine Fahrlehrerin wird der Fahrschueler angelegt und eine Nachricht ueber
+	 * den Erfolg mit der ID des Fahrschuelers wird ausgegeben. Eine beliebige
+	 * Fahrlehrerin wird dem Fahrschueler zugewiesen.
+	 * 
+	 * PRE: Der uebergebene Name und die Anschrift sind "sinnvolle Werte" und
+	 * der Fahrschueler existiert noch nicht.
+	 * 
+	 * @param name
+	 *            Der Name des neuen Fahrschuelers.
+	 * 
+	 * @param anschrift
+	 *            Die Anschrift des neuen Fahrschuelers.
+	 * 
+	 * @return {@link Systemereignis} mit Nachricht ueber Erfolg und eindeutiger
+	 *         ID des neuen Fahrschuelers oder Nachricht ueber Fehlschlag durch
+	 *         keine verfuegbare Fahrlehrerin.
+	 */
+	public Systemereignis fahrschueler_hinzufuegen(String name, String anschrift) {
+		if (this.fahrlehrerin.size()==0) {
+			return(new Systemereignis(Nachricht.Fahrschueler_nicht_hinzugefuegt_keine_Fahrlehrerin_vorhanden));
+		}
+		else {
+			Fahrschueler new_fs = new Fahrschueler(name,anschrift);
+			Random index = new Random();
+			Fahrlehrerindaten fl = this.fahrlehrerin.get(index.nextInt(this.fahrlehrerin.size()));
+			new_fs.set_fahrlehrerin(fl);
+			this.fahrschueler.add(new_fs);
+			Systemereignis systemereignis = new Systemereignis(Nachricht.Fahrschueler_erfolgreich_hinzugefuegt);
+			systemereignis.setID(new_fs.get_ID());
+			return systemereignis;
+		}
+	}
+	
+	/**
+	 * Mit dieser Methode loggt sich die Fahrlehrerin im System ein und ruft
+	 * alle fuer sie eingetragenen Unterrichtsstunden ab. Passt der Name der
+	 * Fahrlehrerin nicht zum Kennzeichen, wird eine entsprechende Fehlermeldung
+	 * ausgegeben. Die gleiche Fehlermeldung wird ausgegeben, wenn es keine
+	 * Fahrlehrerin mit dem Namen gibt. Stimmen die Logindaten, wird eine
+	 * Erfolgsmeldung ausgegeben und im Systemereignis wird eine leserlich
+	 * aufbereitete Uebersicht aller Unterrichtsstunden der Lehrerin
+	 * zurueckgegeben. Stimmen die Logindaten, aber existieren keine
+	 * Unterrichtsstunden fuer die Fahrlehrerin wird der Text
+	 * "Keine Unterrichtsstunden fuer diese Fahrlehrerin vorhanden"
+	 * zurueckgeliefert. Der Text wird im Feld Text des Systemerieignisses
+	 * transportiert und NICHT direkt ausgegeben.
+	 * 
+	 * PRE: true.
+	 * 
+	 * @param name
+	 *            Der Name der Fahrlehrerin.
+	 * 
+	 * @param kennzeichen
+	 *            Das Kennzeichen des Autos der Fahrlehrerin.
+	 * 
+	 * @return {@link Systemereignis} mit Nachricht ueber Erfolg und
+	 *         Textdarstellung aller Unterrichtsstunden der Fahrlehrerin oder
+	 *         Nachricht ueber den Fehlschlag durch falsche Logindaten.
+	 */
+	public Systemereignis fahrlehrerin_einloggene_und_unterrichtsstunden_zurueckgeben(String name, String kennzeichen) {
+		if (this.find_fahrlehrerin(name)==null || this.find_fahrlehrerin(name).get_zugewiesenes_auto().get_kennzeichen()!=kennzeichen) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.